contentcuration/contentcuration/models.py

```diff
@@ -1070,7 +1070,12 @@ def save(self, *args, **kwargs):
             if not self.file_size:
                 self.file_size = self.file_on_disk.size
             if not self.file_format:
-                self.file_format_id = os.path.splitext(self.file_on_disk.name)[1]
+                ext = os.path.splitext(self.file_on_disk.name)[1].lstrip('.')
+                try:
+                    self.file_format = FileFormat.objects.get(pk=ext)
+                except ObjectDoesNotExist:
+                    raise ValueError('Files of type `{}` are not supported.'.format(ext))
```

Comment:
@kollivier I agree that graceful error handling is important.  You're thinking something on the front end?  For a first pass, this seemed like an improvement over the `ObjectDoesNotExist` error that was being thrown before, and it also gets tests passing more gracefully.

Reply:
So one thing we need to be careful about is if we raise, and don't catch that error and return an HttpResponse in the calling code, we don't actually return a response to the client at all. (It might turn into a 504 error, but I haven't tested.) Django / DRF views handle this automagically for us, but our own views don't always make sure to return a HttpResponse in case of error. 

My thinking is more along the lines of ensuring we return a a proper HTTP response - e.g. 400 Bad Request, with an error message about an unsupported file type. Then we can look at having the frontend handle that properly. If we were raising before though, that change would be outside of the scope of this PR.